pages/reference/OS/configuration.md

```diff
@@ -0,0 +1,179 @@
+  "ntpServers": "ntp-wwv.nist.gov resinio.pool.ntp.org",
+  "dnsServers": "208.67.222.222 8.8.8.8",
+  "os": {
+    "network": {
+      "connectivity": {
+        "uri": "https://api.balena-cloud.com/connectivity-check",
+        "interval": "300",
+        "response": "optional value in the response"
+      },
+      "wifi": {
+        "randomMacAddressScan": false
+      }
+    },
+    "udevRules": {
+      "56": "ENV{ID_FS_LABEL_ENC}==\"resin-root*\", IMPORT{program}=\"resin_update_state_probe $devnode\", SYMLINK+=\"disk/by-state/$env{RESIN_UPDATE_STATE}\"",
+      "64": "ACTION!=\"add|change\", GOTO=\"modeswitch_rules_end\"\nKERNEL==\"ttyACM*\", ATTRS{idVendor}==\"1546\", ATTRS{idProduct}==\"1146\", TAG+=\"systemd\", ENV{SYSTEMD_WANTS}=\"u-blox-switch@'%E{DEVNAME}'.service\"\nLBEL=\"modeswitch_rules_end\"\n"
+    },
+    "sshKeys": [
+      "ssh-rsa AAAAB3Nza...M2JB balena@macbook-pro",
+      "ssh-rsa AAAAB3Nza...nFTQ balena@zenbook"
+    ]
+  }
+}
+```
+
+## Valid fields
+
+The following fields can be used to modify the behavior of the host OS.
+
+### hostname
+
+(string) The configured hostname of the device, otherwise the device UUID is used.
+
+### persistentLogging
+
+(boolean) Enable or disable persistent logging on the device. Defaults to false.
+
+### country
+
+(string) The country in which the device is operating. Used for setting the WiFi regulatory domain.
+
+### ntpServers
+
+(string) A space-separated list of NTP servers to use for time synchronization. Defaults to `resinio.pool.ntp.org` servers.
+
+### dnsServers
+
+(string) A space-separated list of preferred DNS servers to use for name resolution.
+
+* When `dnsServers` is not defined, or empty, the default value of Google's DNS (8.8.8.8) is used.
+* When `dnsServers` is `"null"` (a string), the system will not alter the dnsServers to the default server as detailed above.
+* When `dnsServers` is defined and not `"null"` it will be considered as such without any additional servers.
+
+### os
+
+An object containing settings that customize the host OS at runtime.
+
+#### network
+
+##### wifi
+
+An object that defines the configuration related to Wi-Fi.
+
+* "randomMacAddressScan" (boolean) Configures MAC address randomization of a Wi-Fi device during scanning
+
+The following example disables MAC address randomization of Wi-Fi device during scanning:
+
+```json
+"os": {
+  "network" : {
+    "wifi": {
+      "randomMacAddressScan": false
+    }
+  }
+}
+```
+
+##### connectivity
+
+An object that defines configuration related to networking connectivity checks.
+
+* "uri" (string) Value of the url to query for connectivity checks. Defaults to `$API_ENDPOINT/connectivity-check`.
+* "interval" (string) Interval between connectivity checks in seconds. Defaults to 3600.
+* "response" (string). If set controls what body content is checked for when requesting the URI. If it is an empty value, the HTTP server is expected to answer with status code 204 or send no data.
+
```

Comment:
Maybe we can add here something like:
> This feature builds on NetworkManager's connectivity check, which is further documented in the connectivity section at: https://developer.gnome.org/NetworkManager/stable/NetworkManager.conf.html

Reply:
Is it worth mentioning (or linking to other docs) what happens if the connectivity check fails?

Reply:
I've left this with the link to https://developer.gnome.org/NetworkManager/stable/NetworkManager.conf.html for more in-depth discussion.